src/components/organisms/videoList/VideoList.tsx

```diff
@@ -2,27 +2,56 @@ import { Grid } from '@mui/material'
 import { IVideoInfo } from '../../../common/interfaces/IVideoInfo'
 import { PageTitle } from '../../atoms/pageTitle/PageTitle'
 import { VideoCard } from '../../molecules/videoCard/VideoCard'
+import { ShowMoreButton } from '../../atoms/buttons/ShowMoreButton'
+import { HireMeButton } from '../../atoms/buttons/HireMeButton'
+import { useState } from 'react'
 
 type VideoListProps = {
   videos: IVideoInfo[]
   pageType: 'genre' | 'role' | 'recent'
 }
 
 export function VideoList({ videos, pageType }: VideoListProps) {
-  const videoComponents = videos.map(video => (
+  const videosPerPage = 3
+  const [visibleVideos, setVisibleVideos] = useState(
+    videos.slice(0, videosPerPage)
+  )
+
+  const handleShowMore = () => {
+    const currentVisibleCount = visibleVideos.length
+    const nextVisibleVideos = videos.slice(
+      currentVisibleCount,
+      currentVisibleCount + videosPerPage
+    )
+    setVisibleVideos((prevVisibleVideos: IVideoInfo[]) => [
+      ...prevVisibleVideos,
+      ...nextVisibleVideos,
+    ])
+  }
```

Comment:
:art:  This looks like it'll work. But just looks like a lot. We can just overwrite the whole list.

```suggestion
  const handleShowMore = () => {
    setVisibleVideos(videos.slice(0, visibleVideos.length + videosPerPage))
  }
```